packages/block-library/src/navigation/view.js

```diff
@@ -79,6 +79,30 @@ window.addEventListener( 'load', () => {
 		button.addEventListener( 'click', toggleSubmenuOnClick );
 	} );
 
+	// Close modal on clicking internal and external links inside modal.
+	const navigationLinks = document.querySelectorAll(
+		'.wp-block-navigation-item__content'
+	);
+
+	navigationLinks.forEach( function ( link ) {
+		// we need to find the specific parent modal for this link
+		// since .close() won't work without an ID in case we have
+		// mutiple navigation menus in a post/page.
+		const modal = link.closest(
```

Comment:
```suggestion
	navigationLinks.forEach( function ( link ) {
	
		// Ignore non-anchor links.
		if ( ! link.getAttribute( 'href' )?.startsWith( '#' ) ) {
			return;
		}
		
		// we need to find the specific parent modal for this link
		// since .close() won't work without an ID in case we have
		// mutiple navigation menus in a post/page.
		const modal = link.closest(
```

We can ignore links that are no anchors.

Reply:
This would also need to update the comment 

// Close modal on clicking internal and external links inside modal.

Reply:
I think this makes sense. The other cases I thought about where:

- What if the link opens in new tab?
- What if the link is slow to load or is broken? 

But I think in both of these cases we still want this behaviour